diff --git a/exporters/ostream/CMakeLists.txt b/exporters/ostream/CMakeLists.txt
index 1d1c20a6..d9e2412d 100644
--- a/exporters/ostream/CMakeLists.txt
+++ b/exporters/ostream/CMakeLists.txt
@@ -41,7 +41,7 @@ target_include_directories(
   opentelemetry_exporter_ostream_metrics
   PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>")
 target_link_libraries(opentelemetry_exporter_ostream_metrics
-                      PUBLIC opentelemetry_metrics)
+                      PUBLIC opentelemetry_metrics opentelemetry_resources)
 install(
   TARGETS opentelemetry_exporter_ostream_metrics
   EXPORT "${PROJECT_NAME}-target"

diff --git a/exporters/otlp/CMakeLists.txt b/exporters/otlp/CMakeLists.txt
index 1ab0c7cb..fc808c02 100644
--- a/exporters/otlp/CMakeLists.txt
+++ b/exporters/otlp/CMakeLists.txt
@@ -29,7 +29,7 @@ if(WITH_OTLP_GRPC)
                         PROPERTIES EXPORT_NAME otlp_grpc_client)
   target_link_libraries(
     opentelemetry_exporter_otlp_grpc_client
-    PUBLIC opentelemetry_sdk opentelemetry_ext opentelemetry_proto)
+    PUBLIC opentelemetry_common opentelemetry_ext opentelemetry_proto)

   target_link_libraries(opentelemetry_exporter_otlp_grpc_client
                         PRIVATE gRPC::grpc++)
@@ -57,7 +57,8 @@ if(WITH_OTLP_GRPC)
   target_link_libraries(
     opentelemetry_exporter_otlp_grpc
     PUBLIC opentelemetry_otlp_recordable
-           opentelemetry_exporter_otlp_grpc_client)
+           opentelemetry_exporter_otlp_grpc_client
+    PRIVATE gRPC::grpc++)

   list(APPEND OPENTELEMETRY_OTLP_TARGETS opentelemetry_exporter_otlp_grpc)

@@ -72,7 +73,8 @@ if(WITH_OTLP_GRPC)
   target_link_libraries(
     opentelemetry_exporter_otlp_grpc_log
     PUBLIC opentelemetry_otlp_recordable
-           opentelemetry_exporter_otlp_grpc_client)
+           opentelemetry_exporter_otlp_grpc_client
+    PRIVATE gRPC::grpc++)

   list(APPEND OPENTELEMETRY_OTLP_TARGETS opentelemetry_exporter_otlp_grpc_log)

@@ -86,7 +88,8 @@ if(WITH_OTLP_GRPC)
   target_link_libraries(
     opentelemetry_exporter_otlp_grpc_metrics
     PUBLIC opentelemetry_otlp_recordable
-           opentelemetry_exporter_otlp_grpc_client)
+           opentelemetry_exporter_otlp_grpc_client
+    PRIVATE gRPC::grpc++)

   list(APPEND OPENTELEMETRY_OTLP_TARGETS
        opentelemetry_exporter_otlp_grpc_metrics)
@@ -99,7 +102,7 @@ if(WITH_OTLP_HTTP)
                         PROPERTIES EXPORT_NAME otlp_http_client)
   target_link_libraries(
     opentelemetry_exporter_otlp_http_client
-    PUBLIC opentelemetry_sdk opentelemetry_proto opentelemetry_http_client_curl
+    PUBLIC opentelemetry_common opentelemetry_proto opentelemetry_http_client_curl
            nlohmann_json::nlohmann_json)
   if(nlohmann_json_clone)
     add_dependencies(opentelemetry_exporter_otlp_http_client
