diff --git a/exporters/ostream/CMakeLists.txt b/exporters/ostream/CMakeLists.txt
index 1d1c20a6..d9e2412d 100644
--- a/exporters/ostream/CMakeLists.txt
+++ b/exporters/ostream/CMakeLists.txt
@@ -41,7 +41,7 @@ target_include_directories(
   opentelemetry_exporter_ostream_metrics
   PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>")
 target_link_libraries(opentelemetry_exporter_ostream_metrics
-                      PUBLIC opentelemetry_metrics)
+                      PUBLIC opentelemetry_metrics opentelemetry_resources)
 install(
   TARGETS opentelemetry_exporter_ostream_metrics
   EXPORT "${PROJECT_NAME}-target"

