diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2791bb1..450905d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,12 +47,6 @@ set(CPACK_DEBIAN_PACKAGE_DEPENDS "libboost-system-dev, libboost-test-dev, libssl
 set(CPACK_PACKAGE_CONTACT "maintainers@zilliqa.com")
 set(CPACK_DEBIAN_PACKAGE_MAINTAINER "Members of maintainers@zilliqa.com")
 
-# compiler and linker options
-
-add_compile_options(-Wall)
-add_compile_options(-Werror)
-add_compile_options(-Wextra)
-
 if (THREAD_SANITIZER AND ADDRESS_SANITIZER)
     message(FATAL_ERROR "Cannot use ThreadSanitizer (THREAD_SANITIZER=ON) and AddressSanitizer (ADDRESS_SANITIZER=ON) at the same time")
 endif()
